From: Balint Reczey <balint.reczey@canonical.com>
Date: Fri, 21 May 2021 11:46:56 +0200
Subject: src/test/test-mount-util.c: Skip parts of test-mount-util in LXC

LP: #1878051
---
 src/test/test-mount-util.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/src/test/test-mount-util.c
+++ b/src/test/test-mount-util.c
@@ -19,6 +19,7 @@
 #include "strv.h"
 #include "tests.h"
 #include "tmpfile-util.h"
+#include "virt.h"
 
 TEST(mount_option_mangle) {
         char *opts = NULL;
@@ -129,6 +130,11 @@
         _cleanup_(rm_rf_physical_and_freep) char *tmp = NULL;
         _cleanup_free_ char *subdir = NULL;
 
+        if (detect_container() == VIRTUALIZATION_LXC) {
+                log_notice("Testing in LXC, skipping %s due to LP: #1878051", __func__);
+                return;
+        }
+
         if (geteuid() != 0 || have_effective_cap(CAP_SYS_ADMIN) <= 0) {
                 (void) log_tests_skipped("not running privileged");
                 return;
@@ -187,6 +193,11 @@
                 return;
         }
 
+        if (detect_container() == VIRTUALIZATION_LXC) {
+                log_notice("Testing in LXC, skipping %s due to LP: #1878051", __func__);
+                return;
+        }
+
         pid = fork();
         assert_se(pid >= 0);
 
