From fc1d9c7b0c8f83446928d53d644088235bcac235 Mon Sep 17 00:00:00 2001
From: Susant Sahani <ssahani@vmware.com>
Date: Fri, 10 Apr 2020 15:37:21 +0200
Subject: [PATCH 2/3] network: VXLan - Add support for remote address
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1903300
Origin: upstream, https://github.com/systemd/systemd/pull/15397

---
 src/network/netdev/vxlan.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/network/netdev/vxlan.c b/src/network/netdev/vxlan.c
index c27c9bf512..d49f39e0c9 100644
--- a/src/network/netdev/vxlan.c
+++ b/src/network/netdev/vxlan.c
@@ -45,6 +45,13 @@ static int netdev_vxlan_fill_message_create(NetDev *netdev, Link *link, sd_netli
                         r = sd_netlink_message_append_in6_addr(m, IFLA_VXLAN_GROUP6, &v->group.in6);
                 if (r < 0)
                         return log_netdev_error_errno(netdev, r, "Could not append IFLA_VXLAN_GROUP attribute: %m");
+        } else  if (in_addr_is_null(v->remote_family, &v->remote) == 0) {
+                if (v->remote_family == AF_INET)
+                        r = sd_netlink_message_append_in_addr(m, IFLA_VXLAN_GROUP, &v->remote.in);
+                else
+                        r = sd_netlink_message_append_in6_addr(m, IFLA_VXLAN_GROUP6, &v->remote.in6);
+                if (r < 0)
+                        return log_netdev_error_errno(netdev, r, "Could not append IFLA_VXLAN_GROUP attribute: %m");
         }
 
         if (in_addr_is_null(v->local_family, &v->local) == 0) {
@@ -348,6 +355,11 @@ static int netdev_vxlan_verify(NetDev *netdev, const char *filename) {
         if (!v->dest_port && v->generic_protocol_extension)
                 v->dest_port = 4790;
 
+        if (in_addr_is_null(v->group_family, &v->group) == 0 && in_addr_is_null(v->remote_family, &v->remote) == 0)
+                return log_netdev_warning_errno(netdev, SYNTHETIC_ERRNO(EINVAL),
+                                                "%s: VXLAN both 'Group=' and 'Remote=' cannot be specified. Ignoring.",
+                                                filename);
+
         return 0;
 }
 
-- 
2.25.1

