From 700e0d3d87705a6ba01793d7130bbb8e6edbee16 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 18 May 2020 18:30:18 +0200
Subject: [PATCH 2/7] makefs: log about OOM condition
Bug: https://github.com/systemd/systemd/issues/10179
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1838329
Origin: upstream, https://github.com/systemd/systemd/pull/15836

---
 src/partition/makefs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/partition/makefs.c b/src/partition/makefs.c
index d73d67c4e8..df08a5fea6 100644
--- a/src/partition/makefs.c
+++ b/src/partition/makefs.c
@@ -54,11 +54,11 @@ static int run(int argc, char *argv[]) {
         /* type and device must be copied because makefs calls safe_fork, which clears argv[] */
         type = strdup(argv[1]);
         if (!type)
-                return -ENOMEM;
+                return log_oom();
 
         device = strdup(argv[2]);
         if (!device)
-                return -ENOMEM;
+                return log_oom();
 
         if (stat(device, &st) < 0)
                 return log_error_errno(errno, "Failed to stat \"%s\": %m", device);
-- 
2.25.1

