From e9da62b18af647bfa73807e1c7fc3bfa4bb4b2ac Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 22 Apr 2020 20:34:02 +0200
Subject: [PATCH 2/2] core: make sure to restore the control command id, too
Origin: upstream, https://github.com/systemd/systemd/pull/15546
Bug: https://github.com/systemd/systemd/issues/15356
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1873607

Fixes: #15356
---
 src/core/service.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/core/service.c b/src/core/service.c
index 861d82041a..7d5928e455 100644
--- a/src/core/service.c
+++ b/src/core/service.c
@@ -2834,9 +2834,10 @@ static int service_deserialize_exec_command(
                                 break;
         }
 
-        if (command && control)
+        if (command && control) {
                 s->control_command = command;
-        else if (command)
+                s->control_command_id = id;
+        } else if (command)
                 s->main_command = command;
         else
                 log_unit_warning(u, "Current command vanished from the unit file, execution of the command list won't be resumed.");
-- 
2.25.1

