From 659f85a519f4e33c17079d613603db7c822f17aa Mon Sep 17 00:00:00 2001
From: Susant Sahani <ssahani@vmware.com>
Date: Fri, 10 Apr 2020 15:50:23 +0200
Subject: [PATCH 3/3] networkctl: Add support to display VXLan remote address
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1903300
Origin: upstream, https://github.com/systemd/systemd/pull/15397

---
 src/network/networkctl.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

--- a/src/network/networkctl.c
+++ b/src/network/networkctl.c
@@ -1373,9 +1373,17 @@ static int link_status_one(
                 }
 
                 if (IN_SET(info->vxlan_info.group_family, AF_INET, AF_INET6)) {
+                        const char *p;
+
+                        r = in_addr_is_multicast(info->vxlan_info.group_family, &info->vxlan_info.group);
+                        if (r <= 0)
+                                p = "Remote:";
+                        else
+                                p = "Group:";
+
                         r = table_add_many(table,
                                            TABLE_EMPTY,
-                                           TABLE_STRING, "Group:",
+                                           TABLE_STRING, p,
                                            info->vxlan_info.group_family == AF_INET ? TABLE_IN_ADDR : TABLE_IN6_ADDR,
                                            &info->vxlan_info.group);
                         if (r < 0)
