Bug-Ubuntu: https://bugs.launchpad.net/bugs/1880258
Forwarded: not-needed
Reviewed-By: Dan Streetman <ddstreet@canonical.com>

--- systemd-246.6.orig/src/resolve/resolved-dns-transaction.c
+++ systemd-246.6/src/resolve/resolved-dns-transaction.c
@@ -1049,9 +1049,9 @@ void dns_transaction_process_reply(DnsTr
                         char key_str[DNS_RESOURCE_KEY_STRING_MAX];
                         dns_resource_key_to_string(t->key, key_str, sizeof key_str);
                         t->current_feature_level = DNS_SERVER_FEATURE_LEVEL_UDP;
-                        log_warning("Server returned error %s, mitigating potential DNS violation DVE-2018-0001, retrying transaction with reduced feature level %s.",
-                                    dns_rcode_to_string(DNS_PACKET_RCODE(p)),
-                                    dns_server_feature_level_to_string(t->current_feature_level));
+                        log_debug("Server returned error %s, mitigating potential DNS violation DVE-2018-0001, retrying transaction with reduced feature level %s.",
+                                  dns_rcode_to_string(DNS_PACKET_RCODE(p)),
+                                  dns_server_feature_level_to_string(t->current_feature_level));
                         dns_transaction_retry(t, false /* use the same server */);
                         return;
                 }
