From 8270e3d8ed320b41e14ec8578456228f578f0561 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Sun, 10 May 2020 09:19:29 +0200
Subject: [PATCH] seccomp-util: add new syscalls from kernel 5.6 to syscall
 filter table
Origin: upstream, https://github.com/systemd/systemd/commit/8270e3d8ed320b41e14ec8578456228f578f0561
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libseccomp/+bug/1891810


---
 src/shared/seccomp-util.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/src/shared/seccomp-util.c
+++ b/src/shared/seccomp-util.c
@@ -390,6 +390,7 @@ const SyscallFilterSet syscall_filter_se
                 .value =
                 "lookup_dcookie\0"
                 "perf_event_open\0"
+                "pidfd_getfd\0"
                 "ptrace\0"
                 "rtas\0"
 #ifdef __NR_s390_runtime_instr
@@ -456,6 +457,7 @@ const SyscallFilterSet syscall_filter_se
                 "oldstat\0"
                 "open\0"
                 "openat\0"
+                "openat2\0"
                 "readlink\0"
                 "readlinkat\0"
                 "removexattr\0"
