From ac4e03d45bcf4ad2e570cabdb218e9bac003cc80 Mon Sep 17 00:00:00 2001
From: Jan Palus <atler@pld-linux.org>
Date: Thu, 11 Jun 2020 18:23:28 +0200
Subject: [PATCH] login: filenames in /run/systemd/users are uids
Bug: https://github.com/systemd/systemd/issues/16146
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1944711
Origin: upstream, https://github.com/systemd/systemd/pull/16153

Fixes: #16146
---
 src/login/logind.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

--- a/src/login/logind.c
+++ b/src/login/logind.c
@@ -2,6 +2,7 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#include <sys/types.h>
 #include <unistd.h>
 
 #include "sd-daemon.h"
@@ -30,6 +31,7 @@
 #include "strv.h"
 #include "terminal-util.h"
 #include "udev-util.h"
+#include "user-util.h"
 
 static Manager* manager_unref(Manager *m);
 DEFINE_TRIVIAL_CLEANUP_FUNC(Manager*, manager_unref);
@@ -323,11 +325,18 @@ static int manager_enumerate_users(Manag
 
         FOREACH_DIRENT(de, d, return -errno) {
                 User *u;
+                uid_t uid;
 
                 if (!dirent_is_file(de))
                         continue;
 
-                k = manager_add_user_by_name(m, de->d_name, &u);
+                k = parse_uid(de->d_name, &uid);
+                if (k < 0) {
+                        r = log_warning_errno(k, "Failed to parse filename /run/systemd/users/%s as UID.", de->d_name);
+                        continue;
+                }
+
+                k = manager_add_user_by_uid(m, uid, &u);
                 if (k < 0) {
                         r = log_warning_errno(k, "Failed to add user by file name %s, ignoring: %m", de->d_name);
                         continue;
