From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat, 3 Jul 2021 02:23:11 +0900
Subject: test: check memory pressure more frequently

---
 test/units/testsuite-55.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/test/units/testsuite-55.sh b/test/units/testsuite-55.sh
index 63549c8..b7a84c7 100755
--- a/test/units/testsuite-55.sh
+++ b/test/units/testsuite-55.sh
@@ -21,7 +21,7 @@ fi
 
 rm -rf /etc/systemd/system/testsuite-55-testbloat.service.d
 
-echo "DefaultMemoryPressureDurationSec=5s" >>/etc/systemd/oomd.conf
+echo "DefaultMemoryPressureDurationSec=2s" >>/etc/systemd/oomd.conf
 
 systemctl start testsuite-55-testchill.service
 systemctl start testsuite-55-testbloat.service
@@ -29,16 +29,16 @@ systemctl start testsuite-55-testbloat.service
 # Verify systemd-oomd is monitoring the expected units
 oomctl | grep "/testsuite-55-workload.slice"
 oomctl | grep "1.00%"
-oomctl | grep "Default Memory Pressure Duration: 5s"
+oomctl | grep "Default Memory Pressure Duration: 2s"
 
-# systemd-oomd watches for elevated pressure for 5 seconds before acting.
+# systemd-oomd watches for elevated pressure for 2 seconds before acting.
 # It can take time to build up pressure so either wait 2 minutes or for the service to fail.
 timeout="$(date -ud "2 minutes" +%s)"
 while [[ $(date -u +%s) -le $timeout ]]; do
     if ! systemctl status testsuite-55-testbloat.service; then
         break
     fi
-    sleep 5
+    sleep 2
 done
 
 # testbloat should be killed and testchill should be fine
